### ISSUE TITLE
MongoDB replica set in MONGODB_URI not supported

### ISSUE THREAD

**Issue by pommi** [guardian/alerta#289]
In case of a MongoDB replica `,host2[:port2]` is set in MONGODB_URI like it's defined by MongoDB in their [standard URI connection scheme](https://docs.mongodb.com/manual/reference/connection-string/#standard-connection-string-format):
```
mongodb://[username:password@]host1[:port1][,host2[:port2],...[,hostN[:portN]]][/[database][?options]]
```

For example:
```
mongodb://username:password@database-host1:12345,database-host2:12345/database?replicaSet=rs-database
```

This is currently not supported by alerta and will result in this stacktrace:
```
[ERROR] Exception in worker process
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/arbiter.py", line 557, in spawn_worker
    worker.init_process()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/base.py", line 136, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/util.py", line 357, in import_app
    __import__(module)
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/__init__.py", line 96, in <module>
    db = Mongo()
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/database.py", line 30, in __init__
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/database.py", line 48, in connect
    self.connect()
    app.config['MONGO_PORT'] = uri.port
  File "/app/.heroku/python/lib/python2.7/urlparse.py", line 113, in port
    port = int(port, 10)
ValueError: invalid literal for int() with base 10: '12345,database-host2'
```

This is caused by `uri = urlparse(mongo_uri)` in https://github.com/guardian/alerta/blob/master/alerta/app/database.py, because urlparse is not able to parse the scheme with multiple hosts/ports.

Besides this MongoClient is handling this URI perfectly fine.

An alternative would be to use `pymongo.uri_parser`. But, looking a few lines ahead, it's not possible to map multiple hosts to 1 `app.config['MONGO_HOST']` and 1 `app.config['MONGO_PORT']`. Is setting `app.config['XXX']` really necessary there? I've been looking around in the code, but to me it seems those settings are never used after this.

**Comment by satterly** [comment]
Alerta server version 4.8.12 which includes this change is available for download. Please test and confirm fix. Thanks.

**Comment by pommi** [comment]
Wow, you are awesome! :+1:

**Comment by pommi** [comment]
While trying to test this I ran into the Traceback below. This **also** happens when deploying on a single node (mLab Sandbox) on heroku. I'm deploying on heroku using https://github.com/alerta/heroku-api-alerta. So it installs alerta-server using pip, which means I'm running 4.8.12. When I put `alerta-server==4.8.11` in my `requirements.txt` the deployment on the single node works ok like it did before. I'm trying to figure out how to run alerta from the git repository instead of the version from pip so I can also do some better debugging and probably provide you with a fix.

```
[2016-11-28 11:31:53 +0000] [15] [ERROR] Exception in worker process
Traceback (most recent call last):
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/arbiter.py", line 557, in spawn_worker
    worker.init_process()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/base.py", line 126, in init_process
    self.load_wsgi()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/workers/base.py", line 136, in load_wsgi
    self.wsgi = self.app.wsgi()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/base.py", line 67, in wsgi
    self.callable = self.load()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 65, in load
    return self.load_wsgiapp()
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/app/wsgiapp.py", line 52, in load_wsgiapp
    return util.import_app(self.app_uri)
  File "/app/.heroku/python/lib/python2.7/site-packages/gunicorn/util.py", line 357, in import_app
    __import__(module)
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/__init__.py", line 96, in <module>
    db = Mongo()
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/database.py", line 30, in __init__
    self.connect()
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/database.py", line 57, in connect
    self._create_indexes()
  File "/app/.heroku/python/lib/python2.7/site-packages/alerta/app/database.py", line 61, in _create_indexes
    self._db.alerts.create_index([('environment', ASCENDING), ('resource', ASCENDING), ('event', ASCENDING)], unique=True)
  File "/app/.heroku/python/lib/python2.7/site-packages/pymongo/collection.py", line 1387, in create_index
    self.__create_index(keys, kwargs)
  File "/app/.heroku/python/lib/python2.7/site-packages/pymongo/collection.py", line 1298, in __create_index
    sock_info, cmd, read_preference=ReadPreference.PRIMARY)
  File "/app/.heroku/python/lib/python2.7/site-packages/pymongo/collection.py", line 208, in _command
    read_concern=read_concern)
  File "/app/.heroku/python/lib/python2.7/site-packages/pymongo/pool.py", line 239, in command
    read_concern)
  File "/app/.heroku/python/lib/python2.7/site-packages/pymongo/network.py", line 102, in command
    helpers._check_command_response(response_doc, None, allowable_errors)
  File "/app/.heroku/python/lib/python2.7/site-packages/pymongo/helpers.py", line 205, in _check_command_response
    raise OperationFailure(msg % errmsg, code, response)
OperationFailure: not authorized on monitoring to execute command { createIndexes: "alerts", indexes: [ { unique: true, key: { environment: 1, resource: 1, event: 1 }, name: "environment_1_resource_1_event_1" } ] }
```

**Comment by pommi** [comment]
It looks like you resolved this in 713f19f414a99cc5f984f7db5a4b850d86c707b1

**Comment by satterly** [comment]
Yeah, sorry about that. Glad you noticed the fix.
